flink-table/flink-sql-gateway/src/test/java/org/apache/flink/table/gateway/rest/SqlGatewayRestEndpointITCase.java

```diff
@@ -58,154 +58,139 @@
                                         Collections.emptyList(),
                                         SqlGatewayRestAPIVersion.V0))
-                .isInstanceOf(IllegalArgumentException.class);
-
-        // The header only support V1, send request by V1
+                .satisfies(
+                        FlinkAssertions.anyCauseMatches(
+                                IllegalArgumentException.class,
+                                String.format(
+                                        "The requested version V0 is not supported by the request (method=%s URL=%s). Supported versions are: %s.",
+                                        header12.getHttpMethod(),
+                                        header12.getTargetRestEndpointURL(),
+                                        header12.getSupportedAPIVersions().stream()
+                                                .map(RestAPIVersion::getURLVersionPrefix)
+                                                .collect(Collectors.joining(",")))));
+
+        // The header only supports V0, sends request by V0
         CompletableFuture<TestResponse> specifiedVersionResponse =
                 restClient.sendRequest(
                         serverAddress.getHostName(),
                         serverAddress.getPort(),
-                        header2,
+                        header0,
                         EmptyMessageParameters.getInstance(),
                         EmptyRequestBody.getInstance(),
                         Collections.emptyList(),
-                        SqlGatewayRestAPIVersion.V1);
+                        SqlGatewayRestAPIVersion.V0);
 
-        TestResponse testResponse1 = specifiedVersionResponse.get(5, TimeUnit.SECONDS);
-        assertThat(testResponse1.getStatus()).isEqualTo("V1");
+        TestResponse testResponse1 =
+                specifiedVersionResponse.get(timeout.getSize(), timeout.getUnit());
+        assertThat(testResponse1.getStatus()).isEqualTo("V0");
 
-        // The header only support V1, send request by latest version V1
+        // The header supports V1 and V2, lets the client get the latest version as default
         CompletableFuture<TestResponse> unspecifiedVersionResponse =
                 restClient.sendRequest(
                         serverAddress.getHostName(),
                         serverAddress.getPort(),
-                        header2,
+                        header12,
                         EmptyMessageParameters.getInstance(),
                         EmptyRequestBody.getInstance(),
                         Collections.emptyList());
 
-        TestResponse testResponse2 = unspecifiedVersionResponse.get(5, TimeUnit.SECONDS);
-        assertThat(testResponse2.getStatus()).isEqualTo("V1");
+        TestResponse testResponse2 =
+                unspecifiedVersionResponse.get(timeout.getSize(), timeout.getUnit());
+        assertThat(testResponse2.getStatus()).isEqualTo("V2");
```

Comment:
This part is same to `testDefaultVersionRouting`? If so, I think we can remove here.

nit: BTW, it's better to introduce a case that API is only available in the high version, which is the actual case. With the default version routing, the request works.

Reply:
> This part is same to `testDefaultVersionRouting`? If so, I think we can remove here.
   No, it is to test getting the latest version of all the supported versions. In `RestClient` line 315. 

> nit: BTW, it's better to introduce a case that API is only available in the high version, which is the actual case. With the default version routing, the request works.

Added.